### PR TITLE
[tune] Fix unflatten dict

### LINKED ISSUE #11947: [tune] tune fails with deeply-nested configs due to `unflatten_dict`
### What is the problem?
Tune fails when using >2 levels deep configs.

*Ray version and other system information (Python version, TensorFlow version, OS):* 543f7809a6e8

### Reproduction (REQUIRED)
Add deeply-nested value, such as `"a": {"b": {"c": {"d": tune.choice([1, 2])}}}` to the BOHB config here:: https://github.com/ray-project/ray/blob/396ae0b7c2af310cffdd078570893fc87173eb93/python/ray/tune/examples/bohb_example.py#L53

Then run `python ./python/ray/tune/examples/bohb_example.py`. It fails with:

```
$ python ./python/ray/tune/examples/bohb_example.py
File descriptor limit 256 is too low for production servers and may result in connection errors. At least 8192 is recommended. --- Fix with 'ulimit -n 8192'
2020-11-11 20:38:44,944 INFO services.py:1110 -- View the Ray dashboard at http://127.0.0.1:8265
Traceback (most recent call last):
  File "./python/ray/tune/examples/bohb_example.py", line 87, in <module>
    stop={"training_iteration": 100})
  File "/Users/hartikainen/github/ray-project/ray/python/ray/tune/tune.py", line 416, in run
    runner.step()
  File "/Users/hartikainen/github/ray-project/ray/python/ray/tune/trial_runner.py", line 380, in step
    next_trial = self._get_next_trial()  # blocking
  File "/Users/hartikainen/github/ray-project/ray/python/ray/tune/trial_runner.py", line 479, in _get_next_trial
    self._update_trial_queue(blocking=wait_for_trial)
  File "/Users/hartikainen/github/ray-project/ray/python/ray/tune/trial_runner.py", line 855, in _update_trial_queue
    trial = self._search_alg.next_trial()
  File "/Users/hartikainen/github/ray-project/ray/python/ray/tune/suggest/search_generator.py", line 114, in next_trial
    self._experiment.dir_name)
  File "/Users/hartikainen/github/ray-project/ray/python/ray/tune/suggest/search_generator.py", line 121, in create_trial_if_possible
    suggested_config = self.searcher.suggest(trial_id)
  File "/Users/hartikainen/github/ray-project/ray/python/ray/tune/suggest/bohb.py", line 175, in suggest
    return unflatten_dict(config)
  File "/Users/hartikainen/github/ray-project/ray/python/ray/tune/utils/util.py", line 276, in unflatten_dict
    item = item[k]
KeyError: 'b'
```

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### PR DESCRIPTION
**Pull request by hartikainen**
## Why are these changes needed?
To support deeply-nested configs for tune.

## Related issue number
Fixes #11947.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by krfricke — APPROVED** [review]
LGTM - thanks!

File: python/ray/tune/utils/util_test.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    def test_output_type(self):
+        in_ = OrderedDict({'a/b': 1, 'c/d': 2, 'e': 3})
+        out = unflatten_dict(in_)
+        assert type(in_) is type(out)
+
+    def test_one_level_nested(self):
+        result = unflatten_dict({'a/b': 1, 'c/d': 2, 'e': 3})
+        assert result == {'a': {'b': 1}, 'c': {'d': 2}, 'e': 3}
+
+    def test_multi_level_nested(self):
```
Comment (hartikainen) [inline python/ray/tune/utils/util_test.py:18]: This test would fail in master due to the >2-level nesting.

**Comment by krfricke** [comment]
@hartikainen can you fix the linter errors? (run `./scripts/format.sh`)

**Comment by hartikainen** [comment]
Ah, should be fixed now!

**Comment by krfricke** [comment]
Great! cc @richardliaw  tests passed, this should be ready to merge

**Comment by richardliaw** [comment]
good to know you're still alive @hartikainen ;)

**Comment by hartikainen** [comment]
@richardliaw barely ;)

**Pull request merged**